### ISSUE TITLE
Check that author details are not exposed to reviewers

### ISSUE THREAD

**Issue by NateWr** [pkp/pkp-lib#4868]
Since the introduction of publications, the use of `PKPSubmissionService::canUserViewAuthor()` has been changed. We need to run a quick check again to make sure that author names are not exposed to reviewers.

## Acceptance Criteria
- [x] Reviewers should not see author details in the submissions list.
- [x] Reviewers should not see author details when the access a submission via the API.
- [x] Editors assigned as reviewers should not see author details in the submissions list.
- [x] Editors assigned as reviewers should not see author details when they access a submission via the API.

**Comment by amandastevens** [comment]
Items 1 and 3 check out. I didn't check the API items.

**Comment by quoideneuf** [comment]
@NateWr While testing this, I found a case where an API user can access listed submissions, but not individual submissions. 

https://gist.github.com/quoideneuf/2de8b17657030685c34aa079918b8a6a

**Comment by NateWr** [comment]
Thanks! Not intended but I can see how it might happen. Can you confirm a few things for me and then I can try to reproduce tomorrow:

1. Is dbarnes a Journal Manager (`ROLE_ID_MANAGER`)? (it will be if you haven't changed any user groups/roles from the test data.)
2. Is dbarnes assigned as a participant to this submission? (you can see this by opening the submission workflow and looking at the Participants list.)
3. If he is assigned as a participant, in what roles is he assigned? (you may need to check each available workflow stage's Participants list.)
4. What's the status of this submission? What stage is it in? Any review assignments? What other participants are assigned?

**Comment by quoideneuf** [comment]
@NateWr I removed dbarnes's journal editor role, but he remained assigned as the journal editor to the submission. As a result, it appears he can see every submission via the list endpoint, but no submission via its own endpoint. So maybe this is an edge case particular to a user who no longer has the role but still has the associations.

**Comment by quoideneuf** [comment]
@NateWr It doesn't look like the author data is being obscured for reviewers. Below is a screenshot showing AMMCRAE is a reviewer for submission 2. The user has no other role than 'reviewer' but can see the author name via the API per the response pasted below

![image](https://user-images.githubusercontent.com/1626547/72374235-dce3b300-36d7-11ea-9719-7429b744246d.png)

```
$ curl -H "Accept: application/json" http://127.0.0.1:8000/index.php/publicknowledge/api/v1/submissions/2?apiToken=$AMCCRAEKEY | python -m json.tool | grep author
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2542  100  2542    0     0   5196      0 --:--:-- --:--:-- --:--:--  5187
            "authorsString": "Catherine Kwantes",
            "authorsStringShort": "Kwantes",
    "urlAuthorWorkflow": "http://127.0.0.1:8000/index.php/publicknowledge/authorDashboard/submission/2",
```

**Comment by NateWr** [comment]
Thanks Brian! I was able to reproduce the problem you described with an editor reassigned as a reviewer who still has the old stage assignment. I think it is a rare case and not a new issue, so I've filed that separately here: https://github.com/pkp/pkp-lib/issues/5396.

I was also able to produce the issue with author information appearing in the API when the current user has a double blind review assignment.

PR:
https://github.com/pkp/pkp-lib/pull/5395
 
Tests:
https://github.com/pkp/ojs/pull/2577
